### ISSUE TITLE
Reset after settings update.

### ISSUE THREAD

**Issue by alkatar21** [jgclark/vscode-todo-highlight#30]
Not too important, but when settings are changed, the previous highlights are not reseted and VSCode needs to be restarted once.

**Comment by jgclark** [comment]
Really? I find switching to a different tab and back to the one that was open is enough to re-load and re-apply the new settings.  Have you tried that?

**Comment by alkatar21** [comment]
To be more precise, if I have 2 windows open side by side  and switch between them it is not enough. If I switch tabs in one window it works.
![grafik](https://user-images.githubusercontent.com/61387986/132066346-e71d7e60-85c8-4b39-a819-b12dc331ba25.png)
I have changed from my configuration with the yellow font to the default, which is applied, but my configuration remains active until i switch Tabs.

This can also be a VSCode limitation? Then this can be closed.

**Comment by jgclark** [comment]
Yes, it's a good point. As I'm not the original author I'm not sure if this is a plugin limit, or a VSCode limit.
Another way around this might be issue the `Toggle highlights` command within a session.

Leaving this open in case others can advise if there's a way to get VSCode to update its view when needed.

**Comment by alkatar21** [comment]
`Toggle highlight` has no effect on this. It only affects the highlights of the active configuration.

I know you're not the original author and I probably have even less knowledge of js and VSCode plugins, but I wanted to document what I noticed.

**Comment by jgclark** [comment]
> `Toggle highlight` has no effect on this. It only affects the highlights of the active configuration.

Out of interest, are you running different workspaces in the two different windows? Or two panes inside the same workspace window?

> I know you're not the original author and I probably have even less knowledge of js and VSCode plugins, but I wanted to document what I noticed.

Thanks.

**Comment by alkatar21** [comment]
> Out of interest, are you running different workspaces in the two different windows? Or two panes inside the same workspace window?

Two panes inside the same workspace window.

**Comment by ygoe** [comment]
If this issue is about the background colours repainted over already painted backgrounds when settings are changed, then this also applies when starting VSCode. I found an already open document with very intense highlights. Opening another document and closing it again fixed the appearance (like switching tabs does if you already have another tab open). My TODO background colour is #ff200040 so it has transparency, and painting it multiple times sums up each time.
